### PR TITLE
Optimization, code quality, and other changes.

### PR DESCRIPTION
**Pull request by piotrulos**
**This is a draft until all changes will be made**

If this mod optimizes fps it should be also optimized to not decrease performance itself.

Tasks:
- [x] Don't foreach in Update() (in this old version it creates too much GC)
- [x] Don't catch things into nothing (invisible error spam also decreases performance)
- [x] NullCheck instead of empty try block.
- [x] Replace OnSave() with better solution (OnSave is in items.txt so it can run after main save file)
- [x] Replace hardcoded 60 frames = 1 sec with proper timer (independed from FPS)

Fixes: 
- [x] Load draw distance value from settings
- [x] Removed unnecessary inherit from MonoBehaviour

### CODE REVIEW

**Comment by Athlon007** [comment]
Switching from foreach to for should in theory save 2x time on each Update(). Overall, I agree on all of that change.

I just don't see, why and with what OnSave() can be replaced?

**Comment by Krutonium** [comment]
>I just don't see, why and with what OnSave() can be replaced?

Unsure what it can be replaced with.

**Comment by piotrulos** [comment]
OnSave() is called very late (when items.txt are saved) so it was meant to be just for saving stuff. So using it to enable stuff back for saving is risky.
Also the savegame playmaker event triggers over 1300 FSMs, and the execution order is pretty much random.

It is possible to hook this method before savegame event is triggered, 
it require to hook in every save spot, but it will be much more safer to undo everything before saving happens.

**Comment by piotrulos** [comment]
Ok this is my list of changes ready to merge.
I tested it few times, no errors on my side, but recommend testing before publishing.
Changelog in first post.

About OnSave, I changed it to PrepareForSaveGame(), and at the very end on onload there is a function that hooks it into all SavePoints. It hooks in state that happens before savegame is triggered. 

So any new things that need to be done before saving starts,put it in PrepareForSaveGame() function

File: KruFPS/KruFPS.cs [hunk 1]
```diff
@@ -211,36 +213,64 @@ public override void ModSettings()
...
+                FERNDALE.EnableDisable(true);
+                FLATBED.EnableDisable(true);
+                GIFU.EnableDisable(true);
+                HAYOSIKO.EnableDisable(true);
+                JONNEZ.EnableDisable(true);
+                KEKMET.EnableDisable(true);
+                RUSKO.EnableDisable(true);
+                CABIN.SetActive(true);
+                KINEMATIC.isKinematic = true;
+
```
Comment (Krutonium) [inline KruFPS/KruFPS.cs:258]: Axles and Car Dynamics (See https://github.com/Krutonium/KruFPS/blob/master/KruFPS/KruFPS.cs#L40 )

**Pull request merged**